client.go

```diff
@@ -20,6 +20,10 @@ type HAProxyClient struct {
 	conn net.Conn
 }
 
+type HAProxy interface {
+	RunCommand(cmd string) (*bytes.Buffer, error)
```

Comment:
I guess it six of one and half dozen another. The `net.Conn` interface is 8 functions which is considerable to implement for just a stub. The benefit though is that if the proxy expands on any of the functions that `net.Conn` supplies the previously implemented stub would be easy to extend.